### PR TITLE
Refactor PPL and Lucene time series response processing to return DataFrames

### PR DESCRIPTION
**Pull request by idastambuk**
Datasource query() method in OpenSearch returns LegacyResponseData **for time series queries**, but DataFrames for all other types. 
When LegacyResponseData is returned, Grafana-data eventually calls [toDataFrame](https://github.com/grafana/grafana/blob/12dc56ad0cb5b6b982007d1498346fb03be05f41/packages/grafana-data/src/dataframe/processDataFrame.ts#L309) method in order for viz panels to be able to handle the data. 

This is just doing that already in OpenSearch for time series queries and adjusting tests accordingly. 
The motivation behind doing this now and here is that we will soon start to [migrate response handling logic to the backend](https://github.com/grafana/oss-plugin-partnerships/issues/194) and it will be useful to compare output in DataFrames between FE and Backend, both in tests and runtime. 

**How to test:** 
After @fridgepoet's comment, Im attaching an exported dashboard - it works with the OpenSearch DS config from plugin provisioning. 
For Lucene changes:
- queries use the OpenSearch (not serverless) datasource from plugin-provisioning
- there is a Lucene metrics row (edit). Run them without, then with the new changes. The queries should be identical.

For PPL changes:
- queries use the OpenSearch Serverless datasource from plugin-provisioning. **NB**: You might have to change the time field from release_date (as it is in plugin-provisioning) to "timestamp" in order for the queries to work. See [here](https://github.com/grafana/opensearch-datasource/pull/188#issuecomment-1578187488) 
- there is a PPL time series row in the dashboard Run them without, then with the new changes. The panels should look identical.

Fixes https://github.com/grafana/opensearch-datasource/issues/185

More info: ElasticSearch went through the same process when migrating the DS to BE, the PR is [here
](https://github.com/grafana/grafana/pull/58636)

[OpenSearch FE & BE non regression-1685963225258.json.zip](https://github.com/grafana/opensearch-datasource/files/11651914/OpenSearch.FE.BE.non.regression-1685963225258.json.zip)

### CODE REVIEW

**Review by fridgepoet — APPROVED** [review]

File: src/specs/OpenSearchResponse.test.ts [hunk 1]
```diff
@@ -1,12 +1,37 @@
...
+function getTimeField(frame: DataFrame): Field {
+  const field = frame.fields[0];
+  if (field.type !== FieldType.time) {
+    throw new Error('first field should be the time-field');
+  }
+  return field;
+}
+
+function getValueField(frame: DataFrame): Field {
+  const field = frame.fields[1];
```
[the remark this answers is not in the document]
Comment (idastambuk) [reply src/specs/OpenSearchResponse.test.ts:24]: We totally could, it's not a lot of extra code, but I copied this from ElasticSearch and I guess I see merit with having the error documented with a "message" ('first field should be the time-field').

**Comment by idastambuk** [comment]
> I think you'll need to export the dashboard JSON; the dashboard won't be shared with everyone. I was able to look into that dashboard data source since I'm the owner of the instance, but I don't think others on the squad will know which data sources to use.
> 
> I confirm that I visually see the same results on v2.6.1 and on your branch.
> 
> Testing notes:
> 
> * may need to specify some time ranges for the Lucene metrics vs PPL time series?
> * For the PPL queries, it seems there were some manual modifications made on the data source. Can you also specify them (or maybe apply them to the data source in plugin provisioning so anyone can get it set up as it is on the instance)?



> I think you'll need to export the dashboard JSON; the dashboard won't be shared with everyone. I was able to look into that dashboard data source since I'm the owner of the instance, but I don't think others on the squad will know which data sources to use.
> 
> I confirm that I visually see the same results on v2.6.1 and on your branch.
> 
> Testing notes:
> 
> * may need to specify some time ranges for the Lucene metrics vs PPL time series?
> * For the PPL queries, it seems there were some manual modifications made on the data source. Can you also specify them (or maybe apply them to the data source in plugin provisioning so anyone can get it set up as it is on the instance)?

Added the json in the description, thanks! I exported it with a now-1y time range, which seems to produce results. As for modifying the datasource, I don't think I did anything, what did you notice? 🤔

File: src/specs/OpenSearchResponse.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fridgepoet) [reply src/specs/OpenSearchResponse.test.ts:24]: Oh ok, forget that then! I get that it is easier to compare with the "original"

**Comment by idastambuk** [comment]
> On my Cloud instance, I'm seeing in the config page that the Index name and Time field name is different for the data source "search-cloud-ds-test-domain"
> 
> Sorry, I know this is a lot but if you want others in the squad to test with your method I think it will be helpful to know which data source to use. Right now is the data source for the queries in the exported dashboard intended for serverless one or the other one (not serverless)? Otherwise maybe set up a dashboard for them in a common place with the data sources already configured and some notes for comparing it to running it locally.
> 
> Also is it possible to export a time range?

Ah, now I remember, yes! I changed the Time field to timestamp. I think the time field in plugin provisioning is wrong, as Im getting an error when saving the datasource. Will create a PR there to fix it.
Also, this is what I see when I import the dashboard: 
Screenshot 2023-06-06 at 10 27 12 AM

So I think it's clear that both datasources should be set up. I'll add a remark in the description though just to make sure.

**Pull request merged**